client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/index.tsx

```diff
@@ -13,7 +22,27 @@ const PatternAssembler: Step = ( { navigation } ) => {
 	const [ footer, setFooter ] = useState< Pattern | null >( null );
 	const [ sections, setSections ] = useState< Pattern[] >( [] );
 	const [ sectionPosition, setSectionPosition ] = useState< number | null >( null );
-	const { goNext, goBack } = navigation;
+	const { goBack, goNext, submit } = navigation;
+	const { setDesignOnSite } = useDispatch( SITE_STORE );
+	const reduxDispatch = useReduxDispatch();
+	const { setSelectedDesign, setPendingAction } = useDispatch( ONBOARD_STORE );
+	const site = useSite();
+	const siteSlug = useSiteSlugParam();
+	const siteId = useSiteIdParam();
+	const siteSlugOrId = siteSlug ? siteSlug : siteId;
+
+	const getDesign = () =>
```

Comment:
We could do something like [here](https://github.com/Automattic/wp-calypso/blob/bc2a07b51366497f9e1c4f5e8097de481b366510/client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/pattern-assembler-preview.tsx#L36).

There are 2 points we need to take care of
* We have to check the data to avoid sending the array containing `null` value
* The pattern id needs to be encoded with the source site